### ISSUE TITLE
EditorConfig naming rules are not applied to properties in anonymous types

### ISSUE THREAD

**Issue by cremor** [dotnet/roslyn#60521]
**Version Used**: 17.1.3

**Steps to Reproduce**:

1. Create a .editorconfig file with naming rules for properties, e.g.:

```
dotnet_naming_symbols.non_field_members.applicable_kinds           = property,method,local_function,event
dotnet_naming_style.pascal_case.capitalization                     = pascal_case
dotnet_naming_rule.non_field_members_pascal_case.symbols           = non_field_members
dotnet_naming_rule.non_field_members_pascal_case.style             = pascal_case
dotnet_naming_rule.non_field_members_pascal_case.severity          = warning
```

2. Write an anonymous type with a lower case property name, e.g.
```cs
var obj = new {
    test = 1
};
```

**Expected Behavior**:
IDE1006 "Naming rule violation" should be shown.

**Actual Behavior**:
No warning is shown.
